### DISCUSSION TITLE
plugin-cloud-storage: 403 forbidden when uploading a file

### DISCUSSION THREAD

**Discussion by orasik** [payloadcms/payload#5593] [Feature Requests & Ideas]
Hi, 
I'm receiving a 403 error when uploading a file to media. Sometimes I see the following error in logs:

```
NoSuchKey: The specified key does not exist. 
```
But other times, nothing is showing in logs. I only see 403 error in the browser console.

## Context:

The pod deployment is connected to S3 via IAM policy. I have validated the connectivity by CLI and writing a small script to upload a file to S3 from the pod. I am not using access keys or access secrets.

### Config

```
export default buildConfig({
  collections: [Users, Connections, Media],
  typescript: {
    outputFile: path.resolve(__dirname, "payload-types.ts"),
  },
  debug: true,
  upload: {
    useTempFiles: true,
  },
  plugins: [
    cloudStorage({
      collections: {
        'media': {
          adapter: s3Adapter({
            config: {
              region: process.env.S3_REGION,
            },
            bucket: process.env.S3_BUCKET,
          }),
        },
        'connections': {
          adapter: s3Adapter({
            config: {
              region: process.env.S3_REGION,
            },
            bucket: process.env.S3_BUCKET,
          }), // see docs for the adapter you want to use
        },
      },
    }),
  ],
});

```

### Media file

```
export const Media: CollectionConfig = {
  slug: "media",
  access: {
    read: () => true,
    create: () => true,
    update: () => true,
    delete: () => true,
  },
  admin: {
    useAsTitle: "name",
  },
  upload: {
    staticURL: "/media",
    staticDir: "media",
    disableLocalStorage: true,
    imageSizes: [
      {
        name: "thumbnail",
        width: 400,
        height: 300,
        position: "centre",
      },
      {
        name: "card",
        width: 768,
        height: 1024,
        position: "centre",
      },
      {
        name: "tablet",
        width: 1024,
        height: undefined,
        position: "centre",
      },
    ],
    adminThumbnail: "thumbnail",
    mimeTypes: ["image/*"],
  },
  fields: [
    {
      name: "alt",
      type: "text",
    },
  ],
};

```

### Console URL that is showing 403

```
'/api/media?locale=en&fallback-locale=null'
```

### Browser
I tried Chrome and Firefox.

**Comment by denolfe** [comment]
Is there any difference if you remove your `config` property altogether?

**Comment by orasik** [comment]
Thanks @denolfe 
Didn't work. Is there a config to enable more verbose logs?

**Comment by denolfe** [comment]
@orasik Not as of now, but it's starting to look like we should add something like that.

The current state of the plugin is great for straight-forward use cases, but once you get into IAM roles, issues in specific configurations become difficult to troubleshoot.

**Comment by sachin-tcd** [comment]
I am having the same issue. Everything works fine in localhost but in production it gives me 403 error. In s3 bucket I have added CORS as well but still having issue.

**Comment by paulpopus** [comment]
This plugin is now being maintained in the [Packages Directory](https://github.com/payloadcms/payload/tree/main/packages/plugin-cloud-storage) of the [Payload Monorepo](https://github.com/payloadcms/payload). This repo will soon be archived and all open issues will be closed. This issue, however, will be transferred over.

**Comment by denolfe** [comment]
Moving to Discussions as feature request for improved logging on this plugin.

**Comment by hkn-wt** [comment]
any updates ?

**Comment by cweseloh** [comment]
If it's helpful to anyone else, I was having a similar problem and it was due to an AWS WAF (Firewall) issue.  The default rules don't seem to like the payload file upload so I had to add an exception.
